calypso/contracts.go

```diff
@@ -68,12 +75,20 @@ func (c *contractWr) Spawn(rst byzcoin.ReadOnlyStateTrie, inst byzcoin.Instructi
 		if err != nil {
 			return nil, nil, errors.New("passed read argument is invalid: " + err.Error())
 		}
-		_, _, wc, _, err := rst.GetValues(rd.Write.Slice())
-		if err != nil {
-			return nil, nil, errors.New("referenced write-id is not correct: " + err.Error())
+		if !rd.Write.Equal(inst.InstanceID) {
+			return nil, nil, errors.New("the read request doesn't reference this write-instance")
 		}
-		if wc != ContractWriteID {
-			return nil, nil, errors.New("referenced write-id is not a write instance, got " + wc)
+		if c.Cost.Value > 0 {
```

Comment:
I'd also like to see a test for this feature.

Reply:
There is a test in TypeScript, but not in go... Should I add tests for this and the `darcID` in go, too?

Reply:
Is the TS test going to land in cothority or it's in another repo? If it's only in another repo then I think it's best to have some tests in cothority so that we know we're not breaking anything in the future.

Reply:
Once my two PRs are in (1820 and 1821), I'll send a PR for the cothority-TS, which is currently in my dynasent-repo.